### PR TITLE
Add documentation for Jenkins Converter - Developer Hub

### PR DESCRIPTION
**Pull request by heug**
# Description
Initial draft of documentation for the Jenkins Converter--introduction, instructions on use, current limitations, and supported plug-ins.

# Reasons
The Jenkins Converter web tool will be released as part of the Developer Hub, estimated mid-August. Instructions on the tool page will point to this page for instructions on use.

https://circleci.atlassian.net/browse/CIRCLE-28142
https://circleci.atlassian.net/browse/MKTG-1841

### CODE REVIEW

**Comment by heug** [comment]
cc: @michelle-luna @thomasr8

File: jekyll/_cci2/jenkins-converter.md
Comment (michelle-luna) [inline jekyll/_cci2/jenkins-converter.md:48]:
```suggestion
[CircleCI Workflows](https://circleci.com/docs/2.0/workflows/) (the equivalent of Jenkins pipelines) should be fully transferred from your Jenkinsfile to config.yml, including branch filters. The converter will not transfer over any [scheduled builds](https://circleci.com/docs/2.0/configuration-reference/#triggers) to prevent unintentional builds from being triggered.
```

File: jekyll/_cci2/jenkins-converter.md
Comment (michelle-luna) [inline jekyll/_cci2/jenkins-converter.md:52]:
```suggestion
Many of the configuration options within CircleCI jobs don't have equivalents to Jenkins' offerings. It is best practice to start with the following features to get a richer experience from CircleCI:
```

File: jekyll/_cci2/jenkins-converter.md
Comment (michelle-luna) [inline jekyll/_cci2/jenkins-converter.md:62]:
```suggestion
While the Jenkinsfile Converter attempts to directly translate steps, it does not provide full translation of all steps. To address this, the `JFC_STACK_TRACE` key was added to translate specific steps within the output YAML and to provide some guidance on how to proceed with unsupported step directives.
```

File: jekyll/_cci2/jenkins-converter.md
Comment (michelle-luna) [inline jekyll/_cci2/jenkins-converter.md:82]:
```suggestion
**Note: Jenkinsfiles relying on plugins not listed below cannot be converted**. Please remove stanzas relying on those unsupported plugins (for example `options`), otherwise <u>you will see an error message saying something is "Unknown" or "Invalid"</u>. Please submit a ticket with our support center if you have a request to add a plugin to the list.
```

File: jekyll/_cci2/jenkins-converter.md
Comment (michelle-luna) [inline jekyll/_cci2/jenkins-converter.md:165]:
```suggestion
To share any general feedback regarding this project, submit a ticket with our CircleCI Support team.
```

File: jekyll/_cci2/jenkins-converter.md
Comment (michelle-luna) [inline jekyll/_cci2/jenkins-converter.md:13]:
```suggestion
Currently, the converter only supports declarative Jenkinsfiles. While the number of supported plug-ins and steps continue to be expanded, the hope is that this tool gets you started at least 50% of the way, and makes it easier for you to get started on building on CircleCI.
```

File: jekyll/_cci2/jenkins-converter.md
Comment (michelle-luna) [inline jekyll/_cci2/jenkins-converter.md:17]:
```suggestion
* A limited number of syntaxes and plugins are supported. Jenkinsfiles relying on unsupported syntaxes and plugins cannot be converted. Please manually remove them.
```

File: jekyll/_cci2/jenkins-converter.md
Comment (michelle-luna) [inline jekyll/_cci2/jenkins-converter.md:19]:
```suggestion
* Only `Default` is supported as a tool name for `maven`, `jdk` and `gradle` in the [`tools` block](https://www.jenkins.io/doc/book/pipeline/syntax/#tools), and other names will cause conversion failures. Please configure them as follows or remove them manually.
```

File: jekyll/_cci2/jenkins-converter.md
Comment (michelle-luna) [inline jekyll/_cci2/jenkins-converter.md:40]:
```suggestion
A static Docker executor, [cimg/base](https://github.com/CircleCI-Public/cimg-base), is inserted as the [executor](https://circleci.com/docs/reference-2-1/#executors) regardless of the one defined within the Jenkinsfile input.
```

File: jekyll/_cci2/jenkins-converter.md
Comment (michelle-luna) [inline jekyll/_cci2/jenkins-converter.md:42]:
```suggestion
Given that `cimg/base` is a very lean image, it's highly likely that your project will require a different executor. [CircleCI's convenience images](https://circleci.com/docs/2.0/circleci-images/) are a good place to find other images. Refer to [custom Docker image](https://circleci.com/docs/2.0/custom-images/) for advanced steps to create your own custom image.
```

**Comment by heug** [comment]
@michelle-luna notification that you'd reviewed this skipped my inbox :|. thank you for the review!

File: jekyll/_cci2/jenkins-converter.md
Comment (rosieyohannan) [inline jekyll/_cci2/jenkins-converter.md:9]:
```suggestion
```

**Comment by heug** [comment]
@rosieyohannan thank you for the fix! could you review/approve when you get a chance?

File: jekyll/_cci2/jenkins-converter.md
Comment (rosieyohannan) [inline jekyll/_cci2/jenkins-converter.md:10]:
```suggestion
The Jenkins Converter is a web tool that allows you to easily convert a Jenkinsfile to a CircleCI config.yml, helping you to get started building on CircleCI quickly and easily. It can be accessed at https://circleci.com/developer/tools/jenkins-converter.
```

File: jekyll/_cci2/jenkins-converter.md
Comment (rosieyohannan) [inline jekyll/_cci2/jenkins-converter.md:12]:
```suggestion
Currently, the converter only supports declarative Jenkinsfiles. While the number of supported plug-ins and steps continue to be expanded, the hope is that this tool gets you started at least 50% of the way, and makes it easier for you to get started building on CircleCI.
```

File: jekyll/_cci2/jenkins-converter.md
Comment (rosieyohannan) [inline jekyll/_cci2/jenkins-converter.md:41]:
```suggestion
Given that `cimg/base` is a very lean image, it's highly likely that your project will require a different image. [CircleCI's convenience images](https://circleci.com/docs/2.0/circleci-images/) are a good place to find other images. Refer to [custom Docker image](https://circleci.com/docs/2.0/custom-images/) for advanced steps to create your own custom image.
```

Comment (rosieyohannan) [reply jekyll/_cci2/jenkins-converter.md:41]: I wonder if actually instead of linking to https://circleci.com/docs/2.0/circleci-images/ here we should link to the new developer hub images listing?

File: jekyll/_cci2/jenkins-converter.md
Comment (rosieyohannan) [inline jekyll/_cci2/jenkins-converter.md:43]:
```suggestion
Depending on the use case, you might require the [machine executor](https://circleci.com/docs/2.0/executor-types/#using-machine) if your application requires full access to OS resources and the job environment, or the [macOS executor](https://circleci.com/docs/2.0/executor-types/#using-macos).
```

File: jekyll/_cci2/jenkins-converter.md
Comment (rosieyohannan) [inline jekyll/_cci2/jenkins-converter.md:47]: Does this mean workflows are transferred by our Jenkinsfile converter? Or that people need to do this manually?

File: jekyll/_cci2/jenkins-converter.md
Comment (rosieyohannan) [inline jekyll/_cci2/jenkins-converter.md:81]:
```suggestion
**Note: Jenkinsfiles relying on plugins not listed below cannot be converted**. Please remove stanzas relying on those unsupported plugins (for example `options`), otherwise **you will see an error message saying something is "Unknown" or "Invalid"**. Please submit a ticket with our support center if you have a request to add a plugin to the list.
```

**Comment by rosieyohannan** [comment]
Thanks so much for this @heug! I have made a few comments. Also, need to think about how to make this doc accessible. I guess a link here? https://circleci.com/developer/tools/jenkins-converter and also we can link to it from the Migration section of docs? https://circleci.com/docs/2.0/migration-intro/#section=getting-started

File: jekyll/_cci2/jenkins-converter.md
Comment (heug) [reply jekyll/_cci2/jenkins-converter.md:41]: great idea, changed the link

File: jekyll/_cci2/jenkins-converter.md
Comment (heug) [reply jekyll/_cci2/jenkins-converter.md]: they're transferred--i committed a change to make this more explicit

**Comment by heug** [comment]
@rosieyohannan agreed on linking to this doc from the converter itself! planning on linking once we're able to merge this PR. I'll start looking at adding this to the migration page. thanks

**Review by rosieyohannan — APPROVED** [review]
🎉

**Pull request merged**
